chia/wallet/trading/offer.py

```diff
@@ -393,3 +408,23 @@ def from_bytes(cls, as_bytes: bytes) -> "Offer":
         # Because of the __bytes__ method, we need to parse the dummy CoinSpends as `requested_payments`
         bundle = SpendBundle.from_bytes(as_bytes)
         return cls.from_spend_bundle(bundle)
+
+
+def try_offer_decompression(offer_bytes: bytes) -> Offer:
+    try:
+        return Offer.from_compressed(offer_bytes)
+    except TypeError:
+        pass
+    return Offer.from_bytes(offer_bytes)
```

Comment:
Do we need to support uncompressed offers?

Reply:
Maybe not but I also don't see why not.  Maybe someone makes their own software but chooses not to implement the compression for whatever reason.